crates/starknet-types-core/benches/criterion_hashes.rs

```diff
@@ -0,0 +1,108 @@
+use criterion::{BatchSize, BenchmarkId, black_box, Criterion, criterion_group, criterion_main};
+use rand::{RngCore, SeedableRng};
+use rand_chacha::ChaCha8Rng;
+
+use starknet_types_core::felt::Felt;
+use starknet_types_core::hash::{Pedersen, Poseidon};
+use starknet_types_core::hash::StarkHash;
+
+const SEED: u64 = 3;
+
+fn rnd_felt(rng: &mut ChaCha8Rng) -> Felt {
+    let mut felt: [u8; 32] = Default::default();
+    rng.fill_bytes(&mut felt);
+    Felt::from_bytes_be(&felt)
+}
+
+fn rnd_felts(rng: &mut ChaCha8Rng, n: usize) -> Vec<Felt> {
+    (0..n).map(|_| rnd_felt(rng)).collect()
+}
+
+fn rnd_pair_of_felts(rng: &mut ChaCha8Rng) -> (Felt, Felt) {
+    (rnd_felt(rng), rnd_felt(rng))
+}
+
+fn poseidon_hash_single_input(c: &mut Criterion) {
+    let mut rng = ChaCha8Rng::seed_from_u64(SEED);
+    let mut group = c.benchmark_group("Poseidon");
+
+    let (x, y) = rnd_pair_of_felts(&mut rng);
+
+    group.bench_function("hash - single input", |bench| {
+        bench.iter(|| black_box(Poseidon::hash(&x, &y)))
+    });
+}
```

Comment:
not sure how useful is this one

Reply:
Eventually not very useful, I wanted to keep both single and multiple versions in order to eliminate measuring benchmarking artifacts.

Reply:
Btw, there is a 25% difference between single and multiple versions:
![image](https://github.com/starknet-io/types-rs/assets/190855/72117ee2-db3d-4207-a9ce-fcf48c5123e5)
I attribute it to the fact the single version probably benefits from some low level caching. Happy to hear an alternative explanation.

Reply:
I think i'd remove it then

Reply:
done